tests/virtualization/universal/waitfor_guests.pm

```diff
@@ -38,10 +38,23 @@ sub run {
     assert_script_run "cat /etc/hosts";
 
     # Wait for guests to announce that installation is complete
-    script_retry("test -d /tmp/guests_ip", retry => 15, delay => 120);
-    foreach my $guest (@guests) {
-        script_retry("test -f /tmp/guests_ip/$guest", retry => 20, delay => 120);
-        record_info("$guest installed", "Guest installation completed");
+    my $retry = 35;
+    my $count = 0;
+    while ($count++ < $retry) {
+        my @wait_guests = ();
+        foreach my $guest (@guests) {
+            if (script_run("test -f /tmp/guests_ip/$guest") ne 0) {
```

Comment:
Actually it is a bit redundent that we check all the guests each time, those passed guest can be skipped. But it does not matter, the codes work anyway, it fits our target. LGTM.

Reply:
Good observation Julie, thanks for the comment. I was thinking of making a second array with installed guests and re checking only the missing ones, but in this case checking all guest is not slowing down the test time and code is cleaner/easier to read.